File: alzir/Entregável #1 - Algoritmos/JavaScript/mdc.js
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

const mdc = (input) => {
  let [a, b] = input.split(" ");

  if (isNaN(a) || isNaN(b)) {
    console.log("Os valores devem ser números positivos!");
    return;
  }

  if (a <= 0 || b <= 0) {
    console.log("Os números devem ser positivos!");
    return;
  }

  while (b !== 0) {
    let temp = b;
    b = a % b;
    a = temp;
  }
  return console.log(a);
};

readline.question("Digite dois números inteiros: ", (input) => {
  readline.close();
  mdc(input.replace(",", ""));
});

File: alzir/Entregável #1 - Algoritmos/Java/src/Mdc.java
import java.util.Scanner;

public class Mdc {

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.print("Digite dois números inteiros: ");
    String input = sc.nextLine();
    sc.close();
    mdc(input.replace(",", ""));
  }

  public static void mdc(String input) {
    String[] numeros = input.split(" ");

    if (!numeros[0].matches("[0-9]+") || !numeros[1].matches("[0-9]+")) {
      System.out.println("Os valores devem ser números positivos!");
      return;
    }

    int a = Integer.parseInt(numeros[0]);
    int b = Integer.parseInt(numeros[1]);

    if (a <= 0 || b <= 0) {
      System.out.println("Os números devem ser positivos!");
      return;
    }

    while (b != 0) {
      int temp = b;
      b = a % b;
      a = temp;
    }
    System.out.println(a);
  }
}
